lib/linters.py

```diff
@@ -0,0 +1,612 @@
+# Copyright 2008 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from collections import namedtuple, defaultdict
+from copy import deepcopy
+import datetime
+import json
+import re
+import yaml
+from lib import nacaddr
+
+__author__ = 'mikeelkin2@fb.com (Mike Elkin)'
+
+Error = namedtuple('Error', ['filename', 'lineno', 'offset', 'severity',
+               'message', 'category'])
+
+
+def sanitizeNetworkItem(x):
+  # Remove comments and identify just the content we care about
+  return x.split('#', 1)[0].strip()
+
+
+class severity(object):
+  DISABLED = 'disabled'
+  ADVICE = 'advice'
+  WARNING = 'warning'
+  AUTOFIX = 'autofix'
+  ERROR = 'error'
+
+  @classmethod
+  def human(cls, k):
+    mapping = {
+      cls.DISABLED: 'Disabled',
+      cls.ADVICE: 'Advice',
+      cls.WARNING: 'Warning',
+      cls.ERROR: 'Error',
+      cls.AUTOFIX: 'Auto-Fix',
+    }
+    return mapping[k]
+
+  @classmethod
+  def int(cls, k):
+    mapping = {
+      cls.DISABLED: 10,
+      cls.ADVICE: 20,
+      cls.WARNING: 30,
+      cls.ERROR: 40,
+      cls.AUTOFIX: 25,
+    }
+    return mapping[k]
+
+
+class LintErrors(object):
+  def __init__(self):
+    self.errs = []
+    self.filename = None
+
+  def __iter__(self):
+    for e in self.errs:
+      yield e
+
+  def add(self, severity, msg, lineno=0, offset=0, filename=None,
```

Comment:
Couple of issues here, this and the other definitions should have docstrings.
You are reusing severity. You probably want to still rename this to something else to avoid confusion. severity_code?